### ISSUE TITLE
Shuffle true with validation set takes pre-shuffled data for validation set

### ISSUE THREAD

**Issue by jasonmayes** [tensorflow/tfjs#5976]
**System information**
- OS Platform and Distribution : Appears agnostic to platform - happens on all platforms.
- TensorFlow.js installed from (npm or script link): Script
- TensorFlow.js version: 3.11.0
- Browser version: Latest Chrome

**Describe the current behavior**

Currently when calling model.fit with shuffle set to true and also setting aside some of the data to use as validation data a simple multi layered perceptron trained on MNIST data overfits with validation accuracy staying close to 0. Have tried with 20% validation ratio. When I change the ratio to 5% it works fine. This suggests to me that there is a bug where by the validation data is being set aside PRIOR to the shuffle of the training data, which is not working as expected.

**Describe the expected behavior**
The expectation of the user is that the training data will be shuffled in case it is in some order and the validation split will be taken from that shuffled data after the shuffle has taken place.

**Standalone code to reproduce the issue**

Working demo here. For now I have worked around the issue by MANUALLY shuffling the training data arrays before I convert them to Tensors (using tf.util.shuffleCombo(INPUTS, OUTPUTS)). IF you remove this single line of code (on line 27), you will see the error explained above.

https://glitch.com/edit/#!/tensorflowjs-mnist-classifier?path=script.js%3A59%3A0

**Comment by nileshau-afk** [comment]
Hello @jasonmayes  ,

Thank you for bringing this to our attention and for providing a comprehensive overview. We appreciate the time you invested in creating this issue.

Currently, our internal development team is concentrated on other scheduled milestones, so we cannot prioritize this task immediately. However, we would love to see this move forward as a community-led effort.

If you’d like to submit a fix, please fork the repo and open a PR following the [TensorFlow Contribution Guidelines](https://www.tensorflow.org/community/contribute), [Tfjs Contributions ](https://github.com/tensorflow/tfjs/blob/master/tfjs-vis/CONTRIBUTING.md#how-to-contribute)and be sure to link this issue in your PR to avoid duplicate work, kindly leave a comment here if you plan to take this on.

Thank you again for your interest and time.

**Comment by Varun-S10** [comment]
Hi @jasonmayes, Thank you for opening this issue. Since this issue has been open for a long time, we are closing this issue for now.
If this is still relevant and you would like to work on it, please feel free to reopen the issue or submit a pull request referencing it.

Thank you for your understanding.
